src/core/account.js

```diff
@@ -66,8 +65,8 @@ function Account(remote, account) {
   this.on('removeListener', listenerRemoved);
 
   function attachAccount(request) {
-    if (self._account.is_valid() && self._subs) {
-      request.addAccount(self._account_id);
+    if (isValidAddress(self._address) && self._subs) {
```

Comment:
Might be a good idea, but not in this pull request because we should avoid/minimize behavioral changes in refactoring pull requests.